upload.py

```diff
@@ -54,6 +56,34 @@ def parse_components_input() -> list[Component]:
     return [split_component_str(component) for component in re.split('[;\n]', dirs_str) if component.strip()]
 
 
+def set_oidc_token() -> None:
+    github_oidc_url = os.getenv('ACTIONS_ID_TOKEN_REQUEST_URL')
+    github_token_request = os.getenv('ACTIONS_ID_TOKEN_REQUEST_TOKEN')
+    registry_url = os.getenv('IDF_COMPONENT_REGISTRY_URL') or 'https://components.espressif.com'
+
+    if not github_oidc_url or not github_token_request:
+        print('Error: Please set the permissions "id-token: write" in the Github Action')
+        sys.exit(1)
+
+    response = requests.get(
+        f'{github_oidc_url}&audience={registry_url}',
+        headers={'Authorization': f'Bearer {github_token_request}'},
+        timeout=5,
+    )
+
+    if not response.ok:
+        print(f'Error fetching OIDC token: {response.text}')
+        sys.exit(1)
```

Comment:
`requests.get` can fail for more reasons, not only returning not 200.

Furthermore, only `main` should call `sys.exit` and other functions should raise exceptions (like FatalError) instead.

I would suggest to handle the case:

```
try:
    response = requests.get(
        f'{github_oidc_url}&audience={registry_url}',
        headers={'Authorization': f'Bearer {github_token_request}'},
        timeout=5,
    )
   
    response.raise_for_status()

except requests.exceptions.RequestException as e:
    raise FatalError(f"Failed to fetch OIDC token due to a request error.") from e
```